### ISSUE TITLE
Restructuring option can mess up style ordering (and therefore how it is interpreted)

### ISSUE THREAD

**Issue by markstickley** [jakubpawlowicz/clean-css#681]
Minify this with restructuring on:

````
/* http://meyerweb.com/eric/tools/css/reset/
   v2.0 | 20110126
   License: none (public domain)
*/
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed,
figure, figcaption, footer, header, hgroup,
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure,
footer, header, hgroup, menu, nav, section {
  display: block;
}

body {
  line-height: 1;
}

ol, ul {
  list-style: none;
}

blockquote, q {
  quotes: none;
}

blockquote:before, blockquote:after,
q:before, q:after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

/* Screen.css */
body {
  font-family: Helvetica, Verdana, sans-serif;
  margin-bottom: 30px;
}

h1 {
  font-size: 4em;
  font-family: "Oswald", Verdana, sans-serif;
  font-weight: 400;
  text-align: center;
  margin: 20px 0 40px 0;
}

section h1 {
  font-size: 2em;
  font-family: "Oswald", Verdana, sans-serif;
  font-weight: 300;
  margin: 40px 0 15px 0;
}
````

And it sticks the `font-family` directives for `h1, section h1` right at the top -- before the rules for the reset styles are applied, cancelling out the custom font settings.

I'm not convinced this is desirable behaviour! Perhaps it should insert restructured code where the directive first appears; so in this case where the h1 selector was originally? (Just guessing on that, not tested!)

Cheers :)

**Comment by jacksleight** [comment]
Just run in to this issue, can't see why the restructuring behaviour would ever be desirable, it can introduce all sorts of unexpected styling. It should be disabled by default at least.

If you're using gulp-minify-css this is how you turn it off:

```js
minifycss({
    restructuring: false
})
```

**Comment by jakubpawlowicz** [comment]
Looking into it.

**Comment by jakubpawlowicz** [comment]
Fixed in 3.4.5. Thanks for bringing that up!
